### ISSUE TITLE
[BUG] Segmentation fault when using myloader with --innodb-optimize-keys

### ISSUE THREAD

**Issue by yoni-yad2** [mydumper/mydumper#1057]
myloader -V
myloader0.14.2-2, built against MySQL 5.7.40-43 with SSL support with GZIP

myloader -h [RDS MySQL v5.7 instance] -u ... -p ... --innodb-optimize-keys  --enable-binlog --overwrite-tables --source-db mydb
Segmentation fault

Without --innodb-optimize-keys it succeeds.

**Comment by davidducos** [comment]
Can you share the `*-schema.*` files? As it might be possible that the problem is when we parse the content on those files. BTW, are you using [ANSI_QUOTES](https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_ansi_quotes) on the source server?

**Comment by yoni-yad2** [comment]
We are not using ANSI_QUOTES on the source server.

As for the schema files I'm afraid I cannot disclose them but I ran myloader twice for two separate DBs and one of them completed fine and the second got segmentation fault.

The differences between them is that some tables for the erroneous one had tables DDL with

`SOMECOLUMN double DEFAULT '0' COMMENT 'bla bla bla bla bla', `

and :

`CONSTRAINT `fk_id` FOREIGN KEY (`t_id`) REFERENCES `t` (`id`) ON DELETE NO ACTION ON UPDATE NO ACTION`

**Comment by davidducos** [comment]
Hi @yoni-yad2, that is not enough for me to build a test case. I need at least a testing table which you export and when you try to import it is failing.

**Comment by yoni-yad2** [comment]
Was able to recreate this issue using these objects while trying to :

`CREATE TABLE `a_tbl` (
  `a_id` int(11) unsigned NOT NULL,
  `col1` varchar(4) COLLATE utf8mb4_unicode_ci NOT NULL,
  `col2` int(11) unsigned DEFAULT NULL,
  `col3` varchar(4) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  `col4` int(11) unsigned DEFAULT NULL,
  `col5` varchar(15) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  `col6` double NOT NULL COMMENT 'bla_bla_bla_bla',
  `col7` double NOT NULL COMMENT 'bla_bla_bla_bla',
  `col8` int(11) DEFAULT NULL COMMENT 'הערות',
  `col9` int(11) DEFAULT NULL COMMENT 'הערות',
  `col10` int(11) DEFAULT NULL,
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime DEFAULT NULL ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`a_id`),
  KEY `col4` (`col4`),
  KEY `col8` (`col8`),
  KEY `col9` (`col9`),
  KEY `idx1` (`col1`,`col2`,`col3`,`col4`,`col5`),
  KEY `idx2` (`col2`),
  KEY `idx3` (`col3`,`col1`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;

CREATE TABLE `p_tbl` (
  `p_id` int(3) unsigned NOT NULL,
  `col1` varchar(100) NOT NULL,
  `col2` varchar(150) DEFAULT NULL,
  `col3` double DEFAULT '0' COMMENT 'הערות',
  `col4` double DEFAULT '0' COMMENT 'bla bla bla bla bla bla ( bla )',
  `col5` double DEFAULT '0',
  `col6` double DEFAULT '0',
  `col7` double DEFAULT '0',
  `col8` double DEFAULT '0',
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime DEFAULT NULL ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`p_id`),
  UNIQUE KEY `p_uk` (`col1`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

CREATE TABLE `c_tbl` (
  `c_id` int(3) unsigned NOT NULL AUTO_INCREMENT,
  `col1` varchar(100) NOT NULL,
  `col2` varchar(150) DEFAULT NULL,
  `r_id` int(3) unsigned NOT NULL DEFAULT '0',
  `col4` double DEFAULT '0' COMMENT 'הערות',
  `col5` double DEFAULT '0' COMMENT 'bla bla bla bla bla bla ( bla )',
  `col6` double DEFAULT '0' COMMENT 'bla bla bla bla bla bla ( bla )',
  `col7` double DEFAULT '0' COMMENT 'bla bla bla bla bla bla ( bla )',
  `col8` double DEFAULT '0' COMMENT 'bla bla bla bla bla bla ( bla )',
  `col9` double DEFAULT '0' COMMENT 'bla bla bla bla bla bla ( bla )',
  `col10` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `col11` datetime DEFAULT NULL ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`c_id`),
  UNIQUE KEY `col1_uk` (`col1`),
  KEY `fk_r_id_idx` (`r_id`),
  CONSTRAINT `fk_r_id` FOREIGN KEY (`r_id`) REFERENCES `p_tbl` (`p_id`) ON DELETE NO ACTION ON UPDATE NO ACTION
) ENGINE=InnoDB AUTO_INCREMENT=100 DEFAULT CHARSET=utf8;


CREATE TABLE `j_tbl` (
  `col1` int(11) DEFAULT NULL,
  `col2` int(11) NOT NULL COMMENT 'שדה 2',
  `col3` int(11) NOT NULL COMMENT 'שדה 3',
  `col4` int(11) DEFAULT NULL,
  `col5` int(11) unsigned DEFAULT NULL,
  `col6` double NOT NULL COMMENT 'bla_bla_bla_bla',
  `col7` double NOT NULL COMMENT 'bla_bla_bla_bla',
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime DEFAULT NULL ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`col2`,`col3`),
  KEY `col3_idx` (`col3`),
  KEY `col5_idx` (`col5`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;`

**Comment by davidducos** [comment]
Hi @yoni-yad2,
I was not able to reproduce your issue. 
Please, get the core dump https://github.com/mydumper/mydumper/wiki/Support and what OS are you using? did you installed from a package? can you share the log?

**Comment by yoni-yad2** [comment]
I'm running the utilities on an Amazon Linux 2 AWS ec2 instance against MySQL RDS v5.7.41 source and target.

I installed 
mydumper0.14.2-2, built against MySQL 5.7.40-43 with SSL support with GZIP
myloader0.14.2-2, built against MySQL 5.7.40-43 with SSL support with GZIP

using the instructions in the git repo:

```
release=$(curl -Ls -o /dev/null -w %{url_effective} https://github.com/mydumper/mydumper/releases/latest | cut -d'/' -f8)
yum install https://github.com/mydumper/mydumper/releases/download/${release}/mydumper-${release:1}.el7.x86_64.rpm
yum install https://github.com/mydumper/mydumper/releases/download/${release}/mydumper-${release:1}.el8.x86_64.rpm
```

Also I ran the commands you posted and re ran myloader but no core file was created at /tmp directory.

**Comment by davidducos** [comment]
Hi @yoni-yad2,
Can you test with the latest pre/released version?

**Comment by yoni-yad2** [comment]
It seems to be solved on v0.14.4-1
